### ISSUE TITLE
[1.7.0.1] depends on a version of npm and pacote which has an ugly bug

### ISSUE THREAD

**Issue by csillag** [meteor/meteor#9940]
Meteor currently depends on a version of NPM and pacote which has an ugly error, which for me, causes every single docker build to fail.

According to the History,

> The npm package has been upgraded to version 5.10.0, and our fork of its pacote dependency has been rebased against version 7.6.1.

The issue in pacote: https://github.com/zkat/pacote/issues/142
The issue in pnm: https://github.com/npm/npm/issues/19989

The fixes has been released in pacote 8.1.6.

The Meteor for for should be updated with the fixes, too.

Thank you.

**Comment by benjamn** [comment]
We've now published the first beta of Meteor 1.7.1, so you should be able to update like so
```sh
meteor update --release 1.7.1-beta.0
```
Unfortunately, the npm version is somewhat tied to the `meteor-tool` version, so a new Meteor release will be necessary here (not just a package update).

**Comment by impactmass** [comment]
We're also seeing this issue in builds of [Reaction](https://github.com/reactioncommerce/reaction) using Meteor v1.7.0.1 with it's npm@5.10.0. Running with npm@5.6.0 give passing builds.

I've tried running `meteor update --release 1.7.1-beta.0` and still got the `npm ERR! write after end` errors. Looks like that is related to @benjamn statement:
```
Unfortunately, the npm version is somewhat tied to the meteor-tool version, so a new Meteor release will be necessary here (not just a package update).
```

**Comment by janjilek** [comment]
Any updates? 1.7.0.3 and 1.7.1-beta.0 is still crashing with "ERR! write after end" in docker build. Is there some workaround? I am not able to release to production anymore and it's really frustrating...

**Comment by csillag** [comment]
For me, the 1.7.1. beta line worked fine, last time I checked.

**Comment by csillag** [comment]
Just checked, and as of now, both 1.7.0.3 and 1.7.1-beta.15 works correctly for me.

**Comment by janjilek** [comment]
I checked it again and it fails every time with latest 1.7.1-beta and 1.7.0.3. When 1.7.1 will be released, please? I am thinking to migrate away from meteor, because there is some kind of issue with every meteor update...

**Comment by csillag** [comment]
Fortunately, 1.7.1-beta.18 seems to work correctly for me.

**Comment by benjamn** [comment]
We'll close this when Meteor 1.7.1 gets to Release Candidate status (soon).

**Comment by morexlt** [comment]
Same as @csillag 1.7.1-beta.18 work fine!

**Comment by janjilek** [comment]
Hello, I now, that I am quite annoying, but do you have any ETA for the release? I am able to build it with ~1% probability. And every build takes 20-30 minutes... So basically I have one machine trying to build new version 24/7.

**Comment by morexlt** [comment]
> Hello, I now, that I am quite annoying, but do you have any ETA for the release? I am able to build it with ~1% probability. And every build takes 20-30 minutes... So basically I have one machine trying to build new version 24/7.

Ohh @janjilek  that's awful. Check [https://github.com/qualialabs/one](https://github.com/qualialabs/one) only for dev. Or install 1.7.1-beta.18 (only in linux not OSX) hope it helps

**Comment by janjilek** [comment]
Thank you, but I need production ready image, I cannot afford to release unstable/beta version. Development is not a problem, I run it without problems on my dev machine.

**Comment by benjamn** [comment]
Update: we're going to fix this problem in Meteor 1.7.0.4, which should be released today or tomorrow. Stay tuned!

**Comment by janjilek** [comment]
Thank you so much!
